File: db/src/main/java/org/delivery/db/userorder/UserOrderEntity.java
package org.delivery.db.userorder;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.SuperBuilder;
import org.delivery.db.BaseEntity;
import org.delivery.db.store.StoreEntity;
import org.delivery.db.userorder.enums.UserOrderStatus;
import org.delivery.db.userordermenu.UserOrderMenuEntity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

/*@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@SuperBuilder
@Entity(name = "user_order")
public class UserOrderEntity extends BaseEntity {

    @Column(nullable = false)
    private Long userId;

    @JoinColumn(nullable = false, name = "store_id")
    @ManyToOne
    private StoreEntity store;        // store로 변경하던지

    @Column(length = 50, nullable = false)
    @Enumerated(EnumType.STRING)
    private UserOrderStatus status;

    @Column(precision = 11, scale = 4, nullable = false)
    private BigDecimal amount;

    private LocalDateTime orderedAt;

    private LocalDateTime acceptedAt;

    private LocalDateTime cookingStartedAt;

    private LocalDateTime deliveryStartedAt;

    private LocalDateTime receivedAt;

    @OneToMany(mappedBy = "userOrder")
    @ToString.Exclude       // ToString() 호출 시 재귀호출 방지
    @JsonIgnore             // ObjectMapper로 Json 파싱 시 재귀호출 방지
    private List<UserOrderMenuEntity> userOrderMenuList;
}*/

File: db/src/main/kotlin/org/delivery/db/userorder/UserOrderEntity.kt
package org.delivery.db.userorder

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import lombok.ToString
import org.delivery.db.store.StoreEntity
import org.delivery.db.userorder.enums.UserOrderStatus
import org.delivery.db.userordermenu.UserOrderMenuEntity
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity(name = "user_order")
class UserOrderEntity (

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long ?= null,

    @field:Column(nullable = false)
    var userId: Long ?= null,

    @field:JoinColumn(nullable = false, name = "store_id")
    @field:ManyToOne
    var store: StoreEntity ?= null,

    @field:Column(length = 50, nullable = false)
    @field:Enumerated(EnumType.STRING)
    var status: UserOrderStatus ?= null,

    @field:Column(precision = 11, scale = 4, nullable = false)
    var amount: BigDecimal ?= null,

    var orderedAt: LocalDateTime ?= null,

    var acceptedAt: LocalDateTime ?= null,

    var cookingStartedAt: LocalDateTime ?= null,

    var deliveryStartedAt: LocalDateTime ?= null,

    var receivedAt: LocalDateTime ?= null,

    @field:OneToMany(mappedBy = "userOrder")
    //@field:ToString.Exclude       // 롬복의 어노테이션이라 동작 안함
    @field:JsonIgnore               // ObjectMapper로 Json 파싱 시 재귀호출 방지
    var userOrderMenuList: MutableList<UserOrderMenuEntity> ?= null
) {
    override fun toString(): String {
        return "UserOrderEntity(receivedAt=$receivedAt, deliveryStartedAt=$deliveryStartedAt, cookingStartedAt=$cookingStartedAt, acceptedAt=$acceptedAt, orderedAt=$orderedAt, amount=$amount, status=$status, store=$store, userId=$userId, id=$id)"
    }
}